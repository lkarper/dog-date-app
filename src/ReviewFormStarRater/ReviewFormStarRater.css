.ReviewFormStarRater__rate-container {
    float: left;
    display: flex;
    flex-direction: row;
}

.ReviewFormStarRater__rate-container:not(:checked) input {
    position:absolute;
    left:-9999px;
}

.ReviewFormStarRater__rate-container:not(:checked) label {
    float: left;
    width:1em;
    overflow:hidden;
    white-space:nowrap;
    cursor:pointer;
    font-size:30px;
    color:#ccc;
}

.ReviewFormStarRater__rate-container:not(:checked) label:before {
    content: 'â˜… ';
}

.ReviewFormStarRater__rate-container input:checked + label {
    color: #ffc700 !important;    
}

.ReviewFormStarRater__rate-container:not(:checked) label:hover,
.ReviewFormStarRater__rate-container:hover label,
.ReviewFormStarRater__rate-container:hover input:checked + label {
    color: #deb217 !important;  
}

.ReviewFormStarRater__rate-container label:hover ~ label,
.ReviewFormStarRater__rate-container:hover label:hover ~ input:checked + label {
    color: #ccc !important;
}

.ReviewFormStarRater__rate-container input:checked + label:hover,
.ReviewFormStarRater__rate-container input:checked + label:hover ~ label,
.ReviewFormStarRater__rate-container input:checked ~ label:hover,
.ReviewFormStarRater__rate-container input:checked ~ label:hover ~ label,
.ReviewFormStarRater__rate-container label:hover ~ input:checked ~ label {
    color: #c59b08;
}

.ReviewFormStarRater__rate-container:not(:checked) label:focus,
.ReviewFormStarRater__rate-container:not(:checked) label:focus ~ label {
    color: #deb217;  
}
.ReviewFormStarRater__rate-container input:checked + label:focus,
.ReviewFormStarRater__rate-container input:checked + label:focus ~ label,
.ReviewFormStarRater__rate-container input:checked ~ label:focus,
.ReviewFormStarRater__rate-container input:checked ~ label:focus ~ label,
.ReviewFormStarRater__rate-container label:focus ~ input:checked ~ label {
    color: #c59b08;
}

.ReviewFormStarRater__rate-container input:focus + label {
    border:  1px solid black;
}
