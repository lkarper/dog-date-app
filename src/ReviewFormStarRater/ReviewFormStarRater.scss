.star-rating {
    display: inline-block;
  }
  
  #star-filled {
    content: '★ ';
    color: #FEDD31;
  }
  
  #star-empty {
    content: '☆ ';
    color: #CCC;
  }
  
  .stars {
    
    display: inline-block;
    position: relative;
    font-size: 0;
    
    > input {
      position:absolute;
      top:-9999px;
      clip:rect(0,0,0,0);
    }
    
    > label {
      font-size: 16px;
      padding: 0 2px;
      width: 1em;
      height: 1em;
      overflow: hidden;
      display: inline-block;
      vertical-align: middle;
      color: #ccc;
      cursor: pointer;
      
      &:before {
        @extend #star-empty;
      }
    }
    
    &.checked {
      
      > input:checked {
        ~ input + label:before {
          @extend #star-empty;
        }
      }
      
      > label {
        
        &:before {
          @extend #star-filled;
        }
      }
    }
    
    &:hover {
      > input:not(:checked) ~ input + label:before {
        @extend #star-filled;
      }
      
      > label:hover ~ input + label:before {
        @extend #star-empty;
      }
    }
  }