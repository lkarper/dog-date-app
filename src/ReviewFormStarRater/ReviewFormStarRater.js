import React, { useState, useEffect } from 'react';
import './ReviewFormStarRater.css';

const ReviewFormStarRater = (props) => {

    const { classSuffix, currentState, setter, legendText } = props;

    const [starRating, setStarRating] = useState(currentState.toString());

    useEffect(() => {
        if (currentState !== starRating) {
            setter(starRating);
        }
    }, [setter, currentState, starRating]);

    return (
        <fieldset className="sub-fieldset">
            <legend>{legendText}</legend>
            <div className="ReviewFormStarRater__rate-container">                
                <input 
                    type="radio" 
                    id={`star1-${classSuffix}`} 
                    name={`rate-${classSuffix}`} 
                    value="1" 
                    checked={starRating === '1'}
                    onChange={(e) => setStarRating(e.target.value)}
                    required
                />
                <label
                    className={starRating && parseInt(starRating) >= 1 && `auto-selected`} 
                    htmlFor={`star1-${classSuffix}`}
                    style={{
                        color: `${starRating && parseInt(starRating) >= 1 ? '#ffc700' : '#ccc'}`
                    }}
                >
                    1 stars
                </label>              
                <input 
                    type="radio" 
                    id={`star2-${classSuffix}`} 
                    name={`rate-${classSuffix}`} 
                    value="2" 
                    checked={starRating === '2'}
                    onChange={(e) => setStarRating(e.target.value)}
                    required
                />
                <label 
                    htmlFor={`star2-${classSuffix}`}
                    style={{
                        color: `${starRating && parseInt(starRating) >= 2 ? '#ffc700' : '#ccc'}`
                    }}
                >
                    2 stars
                </label>
                <input 
                    type="radio" 
                    id={`star3-${classSuffix}`} 
                    name={`rate-${classSuffix}`} 
                    value="3" 
                    checked={starRating === '3'}
                    onChange={(e) => setStarRating(e.target.value)}
                    required
                />
                <label 
                    htmlFor={`star3-${classSuffix}`}
                    style={{
                        color: `${starRating && parseInt(starRating) >= 3 ? '#ffc700' : '#ccc'}`
                    }}
                >
                    3 stars
                </label>
            
                <input 
                    type="radio" 
                    id={`star4-${classSuffix}`} 
                    name={`rate-${classSuffix}`} 
                    value="4" 
                    checked={starRating === '4'}
                    onChange={(e) => setStarRating(e.target.value)}
                    required
                />
                <label 
                    htmlFor={`star4-${classSuffix}`}
                    style={{
                        color: `${starRating && parseInt(starRating) >= 4 ? '#ffc700' : '#ccc'}`
                    }}
                >
                    4 stars
                </label>
            
                <input 
                    type="radio" 
                    id={`star5-${classSuffix}`} 
                    name={`rate-${classSuffix}`} 
                    value="5" 
                    checked={starRating === '5'}
                    onChange={(e) => setStarRating(e.target.value)}
                    required
                />
                <label 
                    htmlFor={`star5-${classSuffix}`}
                    style={{
                        color: `${starRating && parseInt(starRating) === 5 ? '#ffc700' : '#ccc'}`
                    }}
                >
                    5 stars
                </label>                
            </div>
        </fieldset>
    );
}

export default ReviewFormStarRater;
