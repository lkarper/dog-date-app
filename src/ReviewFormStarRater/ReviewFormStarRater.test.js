import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ReviewFormStarRater from './ReviewFormStarRater';

describe('ReviewFormStarRater component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ReviewFormStarRater />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ReviewFormStarRater />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
