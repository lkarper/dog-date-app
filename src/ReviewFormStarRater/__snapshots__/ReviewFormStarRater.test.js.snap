// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewFormStarRater component renders the UI as expected 1`] = `
<ReviewFormStarRater
  classSuffix=""
  currentState=""
  legendText=""
  required={false}
  setter={[Function]}
>
  <fieldset
    className="sub-fieldset"
  >
    <legend />
    <div
      className="ReviewFormStarRater__rate-container"
    >
      <input
        checked={false}
        className="ReviewFormStarRater__input"
        id="star1-"
        name="rate-"
        onChange={[Function]}
        required={false}
        type="radio"
        value="1"
      />
      <label
        className="ReviewFormStarRater__label"
        htmlFor="star1-"
        style={
          Object {
            "color": "#ccc",
          }
        }
      >
        1 star
      </label>
      <input
        checked={false}
        className="ReviewFormStarRater__input"
        id="star2-"
        name="rate-"
        onChange={[Function]}
        required={false}
        type="radio"
        value="2"
      />
      <label
        className="ReviewFormStarRater__label"
        htmlFor="star2-"
        style={
          Object {
            "color": "#ccc",
          }
        }
      >
        2 stars
      </label>
      <input
        checked={false}
        className="ReviewFormStarRater__input"
        id="star3-"
        name="rate-"
        onChange={[Function]}
        required={false}
        type="radio"
        value="3"
      />
      <label
        className="ReviewFormStarRater__label"
        htmlFor="star3-"
        style={
          Object {
            "color": "#ccc",
          }
        }
      >
        3 stars
      </label>
      <input
        checked={false}
        className="ReviewFormStarRater__input"
        id="star4-"
        name="rate-"
        onChange={[Function]}
        required={false}
        type="radio"
        value="4"
      />
      <label
        className="ReviewFormStarRater__label"
        htmlFor="star4-"
        style={
          Object {
            "color": "#ccc",
          }
        }
      >
        4 stars
      </label>
      <input
        checked={false}
        className="ReviewFormStarRater__input"
        id="star5-"
        name="rate-"
        onChange={[Function]}
        required={false}
        type="radio"
        value="5"
      />
      <label
        className="ReviewFormStarRater__label"
        htmlFor="star5-"
        style={
          Object {
            "color": "#ccc",
          }
        }
      >
        5 stars
      </label>
    </div>
  </fieldset>
</ReviewFormStarRater>
`;
